bodo/pandas/plan_optimizer.pyx

```diff
@@ -506,7 +506,11 @@ cdef unique_ptr[CExpression] make_expr(val):
         if val.resolution.nanoseconds == 1:
             return move(make_const_timestamp_ns_expr(val.value))
         else:
-            raise NotImplementedError("Only support ns timestamp resolution currently, not " + str(val.resolution))
+            try:
+                return move(make_const_timestamp_ns_expr(val.as_unit("ns", round_ok=False).value))
+            except ValueError:
+                raise NotImplementedError("Only timestamps representable in nanoseconds are supported.")
+            
```

Comment:
Another PR fixes this issue:
https://github.com/bodo-ai/Bodo/pull/528/files#diff-f8690c8d7b543fbb56fe7e171dc0ef96a78e81ee30462234973260afc22a4873R519